src/Text/Pandoc/Writers/Powerpoint/Output.hs

```diff
@@ -318,6 +359,40 @@ presentationToArchive opts meta pres = do
                      Nothing -> toArchive . BL.fromStrict <$>
                         P.readDataFile "reference.pptx"
 
+  let (referenceLayouts, defaultReferenceLayouts) =
+        (getLayoutsFromArchive refArchive, getLayoutsFromArchive distArchive)
+  let layoutTitles = SlideLayouts { metadata = "Title Slide" :: Text
+                                  , title = "Section Header"
+                                  , content = "Title and Content"
+                                  , twoColumn = "Two Content"
+                                  }
+  layouts <- for layoutTitles $ \layoutTitle -> do
+        let layout = M.lookup (CI.mk layoutTitle) referenceLayouts
+        let defaultLayout = M.lookup (CI.mk layoutTitle) defaultReferenceLayouts
+        case (layout, defaultLayout) of
+          (Nothing, Nothing) ->
+            throwError (PandocSomeError ("Couldn't find layout named \""
+                                         <> layoutTitle <> "\" in the provided "
+                                         <> "reference doc or in the default "
+                                         <> "reference doc included with pandoc."))
+          (Nothing, Just ((element, path, entry) :| _)) -> do
+            P.logOutput (PptxParserWarning ("Couldn't find layout named \""
+                                           <> layoutTitle <> "\" in provided "
+                                           <> "reference doc. Falling back to "
+                                           <> "the default included with pandoc."))
+            pure SlideLayout { slElement = element
```

Comment:
Should these both be PptxParserWarnings? (One is PandocSomeError)

Reply:
No, I've only used a warning when we have a default we can fall back to. If a named layout isn't found in either the provided reference doc or the default reference doc then we can't do the conversion and have to throw an error. (Since the default reference doc does contain layouts with these names, I'd only expect this to happen if it was changed or moved/removed.)

I guess at this point we could fall back to selecting one of the first 4 layouts (the behaviour before this change)? That would give a successful output if the layouts have been renamed in the provided reference doc but left in the right order, and the default reference doc is missing/incorrect.

Reply:
OK, I understand now.  I don't know about the fallback question.  Probably simplest just to have an error as you do here.